stage_chunk/teardown.sh

```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env bash
+
+set -euxo pipefail
+
+if [ -z "${GOOGLE_APPLICATION_CREDENTIALS:-}" ]; then
+  echo "GOOGLE_APPLICATION_CREDENTIALS is not set. Please set it to your service account key file."
+  exit 1
+fi
+
+if [ -z "${GCP_PROJECT:-}" ]; then
+  echo "GCP_PROJECT is not set. Please set it to your Google Cloud project ID."
+  exit 1
+fi
+
+if [ -z "${REGION:-}" ]; then
+  echo "REGION is not set. Please set it to your Google Cloud region."
+  exit 1
+fi
+
+if [ -z "${GCS_BUCKET:-}" ]; then
+  echo "GCS_BUCKET is not set. Please set it to your Google Cloud Storage bucket name."
+  exit 1
+fi
+
+echo "Teardown started..."
+echo "Project ID: ${GCP_PROJECT}"
+echo "Bucket: ${GCS_BUCKET}"
+echo "Region: ${REGION}"
+
+# Delete Cloud Function
+gcloud functions delete trigger_stage_chunk --region=${REGION} --quiet || true
```

Comment:
Looks odd that you do `set -e` but then fence each command with `|| true`.

Reply:
It will return a non-zero error code if the function does not exist, so that's why the `|| true`.